README.md

```diff
@@ -1,12 +1,12 @@
-**Settler** is a metaprogramming tool for Swift used to resolve complex dependency graphs in a way that encourages code separation and cleanliness while maintaining the safety guarantees of the Swift compiler. If an object in your Resolver cannot be resolved due to a missing or circular dependency, Settler will find it and fail compilation of your program.
+**Settler** is a Swift metaprogramming tool used to resolve complex dependency graphs in a way that encourages code separation and cleanliness while maintaining the safety guarantees of the compiler. If an object in your resolver cannot be resolved due to a missing or circular dependency, Settler will find it and bottom out compilation of your program.
 
 ## The basics
 
-When using Settler, you are responsible for defining **Resolvers**. A Resolver is a type that is responsible for creating a _single complex object_ from a collection of dependencies, or `Keys`.
+When using Settler, you define **resolvers**. A resolver is a type responsible for creating a _single type_ from a collection of dependencies, or `Keys`.
```

Comment:
Torn a bit here on `_single type_`. Should this be "single instance"? Or "single instance of a type"? I feel slightly inclined to keep "object" nomenclature (though it might confuse some) because I feel like most of the time the output of a resolver will indeed by a class instance due to its complexity. Other edits are :guy-fiery-chefs-kiss: